mailbox/caching/src/main/java/org/apache/james/mailbox/caching/CachingMessageMapper.java

```diff
@@ -35,6 +35,11 @@ public CachingMessageMapper(MessageMapper underlying, MailboxMetadataCache cache
     }
 
     @Override
+    public Iterator<MessageUid> listAllMessageUids(Mailbox mailbox) throws MailboxException {
```

Comment:
Isn't this an answer? 

```
Because it allows implementations to not load the entire data structure at once, and I don't have Streams (not java-8)
```

I answered it for 23 hours already...